controllers/volumereplicationgroup_controller.go

```diff
@@ -1317,9 +1332,17 @@ func (v *VRGInstance) preparePVCForVRDeletion(pvc *corev1.PersistentVolumeClaim,
 		return nil
 	}
 
-	// Change PV `reclaimPolicy` back to stored state
-	if err := v.undoPVRetentionForPVC(*pvc, log); err != nil {
-		return err
+	// For Async mode, we want to change the retention policy back to delete
+	// and remove the annotation.
+	// For Sync mode, we don't want to set the retention policy to delete as
+	// both the primary and the secondary VRG map to the same volume. The only
+	// state where a delete retention policy is required for the sync mode is
+	// when the VRG is primary.
+	if !(v.instance.Spec.ReplicationState == ramendrv1alpha1.Secondary &&
+		v.instance.Spec.Sync.Mode == ramendrv1alpha1.SyncModeEnabled) {
```

Comment:
Did you mean Async mode here?  If not, then I don't understand when the PV's retention is changed when the Async mode is set.  The same question applies to primary.  IOW, for Async, this `undoPVRetentionForPVC` was called for when the VRG is secondary and primary.  But now it is never called when Mode==Async.  I am sure I am missing something...

Reply:
The logical not in the if condition makes the code within the if block to be executed for the following things

1) VRG is being deleted as primary for Async
2) VRG is being deleted as secondary for Async
3) VRG is being deleted as primary for Sync

Only when VRG is secondary and it is Sync mode, then the if condition would be false (due to the logical not) and hence would not remove the retention tag (i.e. reclaim policy set to retain) on the PV. This means PV would not be deleted upon PVC deletion as part of application cleanup (due to failover and relocation). This helps in avoiding underlying storage deletion due to PV being deleted (as part of failover and relocation).

Reply:
Thanks.  That was confusing and I didn't even see the logical not.